src/View/InstallQuickInput.ts

```diff
@@ -115,45 +180,58 @@ export async function showInstallQuickInput(): Promise<[ToolchainEnv, Toolchain]
       buttons: [updateButton],
       shouldResume: shouldResume
     });
-    if (version instanceof InnerButton) {
+
+    if (state.selectedItem instanceof InnerButton) {
       Logger.info(logtag, 'press the refresh button');
-      return (input: MultiStepInput) => updateBackend(input, state);
+      this.isUpdateBackend = true;
+      return;
     }
-    state.version = version;
-    state.toolchain = toolchains[versions.indexOf(state.version.label)];
-    return undefined;
+
+    this.version = state.selectedItem.label;
+    this.toolchain = toolchains[versions.indexOf(this.version)];
   }
 
-  async function updateBackend(input: MultiStepInput, state: Partial<State>) {
-    if (state.toolchainEnv === undefined || state.toolchainType === undefined) {
+  async updateBackend(input: MultiStepInput, state: Partial<State>) {
+    if (this.toolchainEnv === undefined || this.toolchainType === undefined) {
+      this.error = 'toolchainenv is undefined.';
       throw Error('toolchainenv is undefined.');
     }
 
-    await state.toolchainEnv.prerequisites();
+    await this.toolchainEnv.prerequisites();
 
     // NOTE(jyoung)
     // Prerequisites request shows the password quick input and this input is
     // automatically added to MultiStepInput's steps. If the user clicks
     // 'back' button on quick input, the previous step is shown, and the password
     // input is show at this time, so it must be removed directly from the steps.
     input.steps.pop();
-    return pickVersion(input, state);
-  }
 
-  function shouldResume() {
-    // Could show a notification with the option to resume.
-    return new Promise<boolean>(
-        (resolve, reject) => {
-            // noop
-        });
+    this.isUpdateBackend = false;
   }
 
-  const state = await collectInputs();
-  return new Promise<[ToolchainEnv, Toolchain]>((resolve, reject) => {
-    if (state.toolchainEnv && state.toolchain) {
-      resolve([state.toolchainEnv, state.toolchain]);
-    } else {
-      reject();
+  async collectInputs() {
+    const state = {selectedItem: undefined} as Partial<State>;
+    await MultiStepInput.run(input => this.pickBackend(input, state));
+    await MultiStepInput.run(input => this.pickType(input, state));
+    await MultiStepInput.run(input => this.pickVersion(input, state));
+    while (this.isUpdateBackend) {
+      await MultiStepInput.run(input => this.updateBackend(input, state));
+      await MultiStepInput.run(input => this.pickVersion(input, state));
     }
```

Comment:
Um,, in existing code, it can go to previous step, but it can't. Is it your intention..?

Existing code:
![quick_input_existing](https://user-images.githubusercontent.com/7223627/177930582-e1a87cd1-7d49-464a-a4c8-7c8990ab00b1.gif)

This PR:
![quick_input_suggestion](https://user-images.githubusercontent.com/7223627/177930568-eae25097-aade-4bda-a47c-46fb53d21025.gif)

Reply:
> Um,, the existing code can go to previous step, but this PR can't. Is it your intention..?

No it's not my intention. This is my fault. Thank you for commenting me. I'll fix. Thanks.

Reply:
Fixed